packages/react-native-updater/android/src/main/java/app/capgo/rnupdater/CapgoUpdaterModule.kt

```
@@ -0,0 +1,275 @@
+      try {
+        val id = options.getString("id") ?: throw IllegalArgumentException("id required")
+        val record = BundleStore.get(reactContext, id) ?: throw IllegalStateException("bundle not found")
+        check(id == CapgoConfig.KEY_BUILTIN || BundleStore.jsBundleFile(reactContext, id).exists()) {
+          "bundle files missing"
+        }
+        BundleStore.setCurrent(reactContext, id)
+        BundleStore.setNext(reactContext, null)
+        CapgoHttp.sendStats(reactContext, "set", record.version)
+        CapgoUpdater.reload(reactContext)
+        promise.resolve(bundleToMap(record))
+      } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)